petition_service/internal/service/petition_service.go

```diff
@@ -43,14 +46,17 @@ func (svc *PetitonService) CreateNew(petition models.Petition) (uint, error) {
 }
 
 func (svc *PetitonService) CreateVote(vote models.Vote) error {
-
-	if err := svc.repo.CheckIfExists(vote.PetitionID); err != nil {
-		return err
-	}
-	if err := svc.repo.SaveVote(&vote); err != nil {
-		return err
+	if res := svc.repo.HasUserVoted(vote.UserID, vote.PetitionID); res {
```

Comment:
Here maybe remove the else's, since the if statement returns anyway and they're not necessary.